### DISCUSSION TITLE
Help USB OBD2

### DISCUSSION THREAD

**Discussion by datamaskinen** [collin80/SavvyCAN#492] [Q&A]
Hi,
I have bought a USB to OBD2 cable (cheep chines thing)

My system is:
Kubuntu 22.04.1 LTS
5.15.0-47-generic
x86_64 x86_64 GNU/Linux

In OBD Auto Doctor the cable works :
-OBD Auto Doctor-
ELM327 v1.5 ISO 9141-2

But in Savvycan it gives me following :
SavvyCAN
$ sudo ./SavvyCAN-305dafd-x86_64.AppImage

-connection settings-
-add New device connection-
it finds the usb and say its a GVRET
GVRET - ttyUSB0 - 2 busses - not connected

in console:

Failed to connect to GVRET at that com port
Serial connection to a GVRET device
Created Serial Port Object
Trying Standard Serial Mode
Connecting to GVRET Device!
Write to serial -> e7 e7 f1 c f1 6 f1 7 f1 1 f1 9
Got data from serial. Len = 1
0

$ sudo lsusb

ID 1a86:7523 QinHeng Electronics CH340 serial converter

$ sudo lsmod |grep ch

ch341                  24576  0
usbserial              57344  1 ch341

$ sudo statserial /dev/ttyUSB0

Device: /dev/ttyUSB0

Signal  Pin  Pin  Direction  Status  Full
Name    (25) (9)  (computer)         Name
-----   ---  ---  ---------  ------  -----
FG       1    -      -           -   Frame Ground
TxD      2    3      out         -   Transmit Data
RxD      3    2      in          -   Receive  Data
RTS      4    7      out         1   Request To Send
CTS      5    8      in          0   Clear To Send
DSR      6    6      in          0   Data Set Ready
GND      7    5      -           -   Signal Ground
DCD      8    1      in          0   Data Carrier Detect
DTR     20    4      out         1   Data Terminal Ready
RI      22    9      in          0   Ring Indicator

$ sudo screen /dev/ttyUSB0 38400

>ELM327 v1.5>OK

**Comment by datamaskinen** [comment]
![image](https://user-images.githubusercontent.com/13379633/188304205-e869cfda-d5f6-49bf-b465-fb38dbbbcf44.png)

**Reply by MotorvateDIY** [reply]
That device will never work with SavvyCan, BUT it will work with any ELM327 compatible app like Torque Pro or OBDFusion.
